test/lib/validator.js

```diff
@@ -0,0 +1,520 @@
+'use strict';
+
+// mocha defines to avoid JSHint breakage
+/* global describe, it, before, beforeEach, after, afterEach */
+require('core-js/shim');
+
+var deepEqual = require('../utils/test_utils').deepEqual;
+var validator = require('../../lib/validator');
+var P = require('bluebird');
+
+function extendSchemaInfo(schema) {
+    // Create summary data on the primary data index
+    schema.iKeys = schema.index.filter(function(elem) {
+        return elem.type === 'hash' || elem.type === 'range'
+    })
+    .map(function(elem) {
+        return elem.attribute;
+    });
+    schema.iKeyMap = {};
+    schema.staticKeyMap = {};
+    schema.index.forEach(function(elem) {
+        if (elem.type === 'static') {
+            schema.staticKeyMap[elem.attribute] = elem;
+        } else {
+            schema.iKeyMap[elem.attribute] = elem;
+        }
+    });
+    return schema;
+}
+
+describe('Unit tests for validation methods', function() {
+    var sampleSchema = {
+        table: 'simple-table',
+        attributes: {
+            key: 'string',
+            tid: 'timeuuid',
+            latestTid: 'timeuuid',
+            range: 'string',
+            body: 'blob'
+        },
+        index: [
+            {attribute: 'key', type: 'hash'},
+            {attribute: 'latestTid', type: 'static'},
+            {attribute: 'tid', type: 'range', order: 'desc'},
+            {attribute: 'range', type: 'range', order: 'desc'}
+        ],
+        tid: 'tid'
+    };
+    sampleSchema = validator.validateAndNormalizeSchema(sampleSchema);
+    // Prepare iKeys and iKeyMap for a sample schema
+    sampleSchema = extendSchemaInfo(sampleSchema);
+
+    describe('Schema validation', function() {
+        it('must have at least one attribute', function() {
+            return P.try(function() {
```

Comment:
No need for Promises in the tests, most calls made here are synchronous, so surrounding calls with `try {} catch {}` blocks suffices.